local-cli/generator-android/templates/src/app/react.gradle

```diff
@@ -32,10 +32,7 @@ gradle.projectsEvaluated {
         buildTypes.each { buildTypeName ->
             // Create variant and source names
             def sourceName = "${buildTypeName}"
-            def targetName = "${sourceName.capitalize()}"
-            if (productFlavorName) {
-                sourceName = "${productFlavorName}${targetName}"
-            }
+            def targetName = "${productFlavorName.capitalize()}${sourceName.capitalize()}"
```

Comment:
this will not work, because when a product flavor does not exist the buildtype should not be capitalized.